docs/src/modules/ROOT/pages/enterprise-edition/enterprise-edition.adoc

```diff
@@ -716,5 +717,215 @@ the host is likely to hang or freeze,
 unless there is an OS specific policy in place to avoid Timefold Solver from hogging all the CPU processors.
 ====
 
+[#automaticNodeSharing]
+=== Automatic node sharing
 
+[NOTE]
+====
+This feature is a commercial feature of Timefold Solver Enterprise Edition.
+It is not available in the Community Edition.
+====
+
+When using xref:constraints-and-score/score-calculation.adoc#constraintStreams[constraint streams], each xref:constraints-and-score/score-calculation.adoc#constraintStreamsBuildingBlocks[building block] forms a node in the score calculation network.
```

Comment:
Think like a user who wants to use this feature. What do they want?

- What do I get out of it?
- How do I enable it?

Everything else is secondary. Start with a simple paragraph explaining what it's for (improves score speeds when many constraints). If you can give a ballpark estimate for the improvements (10%, 50%, ...), do that. Then show the configuration tab. Then mention the downside (debugging).

Only after that, separate sub-sections can explain all the gory details like what it does, what's the downside. Most people won't read that, they just want to get stuff done. They will be happy with the first paragraph. The other paragraphs are for people who like to understand, like us.